src/Interface/Application/TreeViewCollaborators.cc

```diff
@@ -78,9 +80,81 @@ void HideItemsNotMatchingString::operator()(QTreeWidgetItem* item)
 bool HideItemsNotMatchingString::shouldHide(QTreeWidgetItem* item)
 {
   auto text = item->text(0);
-  if (useRegex_)
+  if (searchType_ == SearchType::STARTS_WITH)
+    return !text.startsWith(start_, Qt::CaseInsensitive);
+  else if(searchType_ == SearchType::WILDCARDS
+          || boost::contains(start_, "*"))
     return !match_.exactMatch(text);
-  return !text.startsWith(start_, Qt::CaseInsensitive);
+  else
+    return !fuzzySearchAllPatterns(text, start_);
+}
+
+bool HideItemsNotMatchingString::fuzzySearchAllPatterns(const QString& text, const QString& pattern)
+{
+  std::string pattern_str = removeAllSpecialCharacters(pattern.toStdString());
+  std::vector<std::string> pattern_split;
+  boost::split(pattern_split,
+               pattern_str,
+               [](char c){return c == ' ';});
+
+  // Remove empty vectors
+  for(int i = pattern_split.size()-1; i >= 0; i--)
+  {
+    if(pattern_split[i].empty())
+      pattern_split.erase(pattern_split.begin() + i);
+  }
+
+  // Every word in the pattern must match
+  for(std::string str : pattern_split)
+  {
+    if(!fuzzySearch(text.toStdString(), str))
+    {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool HideItemsNotMatchingString::fuzzySearch(const std::string& text, const std::string& pattern)
+{
+  int patternIndex = 0;
+
+  for(int t = 0; t < text.length(); t++)
+  {
+    bool isUpperChar = pattern[patternIndex] < 97;
+
+    // Counts as a match if letter found. Only case sensitive if search char is upper case
+    if((isUpperChar && (text[t] == pattern[patternIndex]))
+       || (!isUpperChar && (std::tolower(text[t]) == std::tolower(pattern[patternIndex]))))
```

Comment:
On Windows `tolower` needs the `#include <cctype>` header.

Reply:
It's added now.